src/PostTypes/WordpressPost.php

```diff
@@ -1078,6 +1083,19 @@ static function isHierarchical($postType = null) {
 		return is_post_type_hierarchical($postType);
 	}
 
+    private static function getSelfPostTypeConstraint()
+    {
+        // If `get*()` methods are called on `WordpressPost` directly (not a subclass), do not
+        // constrain the type of posts returned unless specified.
+        if (static::class === self::class) {
```

Comment:
This isn't particularly reliable, as there are intermediate classes (e.g. `RoutemasterPost`)

Perhaps make use of `static::$postTypes` instead?

Reply:
I'm not sure that works @rasmuswinter, as we seem to only populate keys of that individually, on-demand, in `postType()`. We also can't just try calling that method currently, because it bails out with `die()` if there is no match. It also doesn't seem like something we could rely on in the general case because child classes are free to override the method with their own implementation.

`RoutemasterPost` is the only abstract class within the scope of this lib. It seems like checking for either `WordpressPost` or `RoutemasterPost` and excluding just those two would be reasonably safe? Especially given that historically, we have had no such check and just attempted to apply the constraint in all cases, for the method which had it.

Reply:
Ah, I was referring to the wrong thing. I meant to refer to `PostTypeManager` instead, which keeps a list of registered oowp post types

Reply:
Got it, thanks @rasmuswinter. https://github.com/outlandishideas/oowp/pull/14/commits/f78b66e0d96c15e48dc077a74dd565c5b56d959e hopefully sorts this for the general case.